session-replay/Sources/DatadogSessionReplay/Writer/Models/SRDataModels.swift

```diff
@@ -20,7 +20,7 @@ internal struct SRSegment: SRDataModel {
     internal let hasFullSnapshot: Bool?
 
     /// The index of this Segment in the segments list that was recorded for this view ID. Starts from 0.
-    internal let indexInView: Int64
+    internal let indexInView: Int64?
```

Comment:
maybe we could add documentation why it could be nil?

Reply:
It's part of the `SRDataModels`, so everything (including the comment) comes from JSON schema and is code-generated. This field was [made optional for Mobile SR purpose](https://github.com/DataDog/rum-events-format/pull/100) - but AFAIK the goal is to provide value for it (_RUMM-2518: Correctly resolve or find different alternative for the `indexInView` segment property_). I'll keep it as it is for now - I'm not sure on the direction we'll take on, but the work is scheduled and tracked in JIRA.